### PR TITLE
Adds in the ability for the client to get a remote_id from the server

### PR DESCRIPTION
**Pull request by sam-phinizy**
Add in the optional ability for a worker to get it's ID from the server using the heartbeat mechanism. This should only be enabled if `PREFECT_EXPERIMENTAL_WORKER_LOGGING_API_ENABLED` is set. This PR doesn't do any logging atm, just set the ground work for future PRs. 

### Checklist

- [ ] This pull request references any related issue by including "closes `<link to issue>`"
  - If no issue exists and your change is not a small fix, please [create an issue](https://github.com/PrefectHQ/prefect/issues/new/choose) first.
- [x] If this pull request adds new functionality, it includes unit tests that cover the changes
- [ ] If this pull request removes docs files, it includes redirect settings in `mint.json`.
- [x] If this pull request adds functions or classes, it includes helpful docstrings.

### CODE REVIEW

File: src/prefect/workers/base.py [hunk 1]
```diff
@@ -724,10 +733,31 @@ async def sync_with_backend(self):
         queues. Sends a worker heartbeat to the API.
         """
         await self._update_local_work_pool_info()
+        # Only do this logic if we've enabled the experiment, are connected to cloud and we don't have an ID.
+        if (
+            get_current_settings().experiments.worker_logging_to_api_enabled
+            and (
+                "api.prefect.cloud" in get_current_settings().api.url
```
Comment (desertaxle) [inline src/prefect/workers/base.py:740]: You can check with the current client if it's connected to Cloud like this:
```
self._client.server_type == ServerType.CLOUD
```

File: src/prefect/client/orchestration.py [hunk 2]
```diff
@@ -2594,22 +2594,36 @@ async def send_worker_heartbeat(
...
                 "heartbeat_interval_seconds": heartbeat_interval_seconds,
-            },
+            }
+            | return_dict,
         )
 
+        if get_worker_id and resp.status_code == 200:
+            return UUID(resp.text)
+        else:
+            return None
```
Comment (desertaxle) [inline src/prefect/client/orchestration.py:2625]: I think we might want to include a check for whether the client is connected to Cloud in this method. If someone is using this method directly with an self-hosted server, it'd cause problems.

**Review by desertaxle — CHANGES_REQUESTED** [review]
This is looking good! Just a few minor comments from me

File: src/prefect/client/orchestration.py (hunk 2 above)
Comment (sam-phinizy) [reply src/prefect/client/orchestration.py:2625]: By default `/heartbeat` should only send 204s not 200s but I agree it's probably better to be safe then sorry!

File: src/prefect/client/orchestration.py [hunk 3]
```diff
@@ -2594,22 +2595,43 @@ async def send_worker_heartbeat(
...
-    ):
+        get_worker_id: bool = False,
+    ) -> Optional[UUID]:
         """
         Sends a worker heartbeat for a given work pool.
 
         Args:
             work_pool_name: The name of the work pool to heartbeat against.
             worker_name: The name of the worker sending the heartbeat.
+            return_id: Whether to return the worker ID.
```
Comment (desertaxle) [inline src/prefect/client/orchestration.py:2606]: I think it'd be worthwhile to add a note here to explain why this kwarg is ignored in some cases.
```suggestion
            return_id: Whether to return the worker ID. Note: will return `None` if the connected server does not support returning worker IDs, even if `return_id` is `True`.
```

**Review by desertaxle — APPROVED** [review]
One small comment on the client method docstring, but the rest LGTM!

**Pull request merged**